packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -836,33 +880,41 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
         const splitButtonContainer = this._splitButtonContainers.get(item.key);
         this._onItemSubMenuExpand(item,
           ((item.split && splitButtonContainer) ? splitButtonContainer : targetElement) as HTMLElement);
+        this._enterTimerId = undefined;
       }, this._navigationIdleDelay);
     } else {
       this._enterTimerId = this._async.setTimeout(() => {
         this._onSubMenuDismiss(ev);
         targetElement.focus();
+        this._enterTimerId = undefined;
       }, timeoutDuration);
     }
   }
 
   private _onItemMouseDown(item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement>) {
-    if (item.onMouseDown) {
+    if (item.onMouseDown && this._enterTimerId === undefined) {
       item.onMouseDown(item, ev);
     }
   }
 
-  private _onItemClick(item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement>) {
+  private _onItemClick(item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>) {
     this._onItemClickBase(item, ev, ev.currentTarget as HTMLElement);
   }
 
   private _onSplitItemClick(item: IContextualMenuItem, ev: React.MouseEvent<HTMLElement>) {
     const splitButtonContainer = this._splitButtonContainers.get(item.key);
     // get the whole splitButton container to base the menu off of
+
+    // TODO don't process anything if we're processing a mouse enter
```

Comment:
Fix

Reply:
Is this comment valid anymore?

Reply:
It's not, we actually get rid of the previous action to process the click event now.